### PR TITLE
Added search history 

### PR DESCRIPTION
**Pull request by archroid**
Added new search history option. Still not completed. Any suggestions for the rest?

### CODE REVIEW

**Comment by IceDBorn** [comment]
Any screenshots?

**Comment by archroid** [comment]
> Any screenshots?

image

**Comment by IceDBorn** [comment]
Seems nice, did you add an option to toggle this though?

**Comment by archroid** [comment]
> Seems nice, did you add an option to toggle this though?

I will

File: app/src/main/java/com/github/libretube/SearchFragment.kt [hunk 1]
```diff
@@ -143,4 +176,36 @@ class SearchFragment : Fragment() {
...
+        }
+
+
+
+        if (splited_history.size > 10) {
+            splited_history = splited_history.takeLast(10)
+        }
+
+
+        sharedPreferences.edit().putString("search_history", splited_history.joinToString("|"))
```
Comment (FireMasterK) [inline app/src/main/java/com/github/libretube/SearchFragment.kt:200]: What if I search for something that contains `|`?

Comment (archroid) [reply app/src/main/java/com/github/libretube/SearchFragment.kt:200]: Yeah you are right
Any suggestions?

Comment (FireMasterK) [reply app/src/main/java/com/github/libretube/SearchFragment.kt:200]: To write the objects: (I haven't tested if this works)
```kotlin
        val baos = ByteArrayOutputStream()
        ObjectOutputStream(baos).use {
            it.writeObject(arrayOf("1", "2", "3", "4", "5"))
        }
        baos.use {
            val encoded = Base64.encodeToString(it.toByteArray(), Base64.DEFAULT)
        }
```

Comment (archroid) [reply app/src/main/java/com/github/libretube/SearchFragment.kt:200]: Will test...
Thx

**Pull request merged**